### DISCUSSION TITLE
Dynamic line-height to make vim/nvim-like app full screen?

### DISCUSSION THREAD

**Discussion by cybershape** [wezterm/wezterm#2000] [Ideas]
Currently wez calculate number of lines/rows based on windows size. When it comes to the bottom of window, if the "left" space is not enough for a full line, then there will be an empty space, which makes vim/nvim not able to do full screen.

Is it possible to adjust the line-height dynamically to make the spare-space distribute evenly between all lines? It won't display more information/lines, but it will make wez the best term for vim/nvim.

**Comment by wez** [comment]
I don't think this would be practical: in pretty much every case, there won't be enough whole pixels to make a practical difference, and the additional fractional pixels are likely to lead to weird vertical positioning issues.

What is a bit more feasible is adjusting the padding to make it more balanced; https://github.com/wez/wezterm/issues/1124 is a request for something like that

**Reply by cybershape** [reply]
That makes a lot of sense.

May I ask what's the smallest fractional pixel we can add to each line without causing any issue ?

**Reply by wez** [reply]
fonts are rasterized to bitmaps consisting of an integer number of pixels. Fractional pixels introduce all kinds of problems with placement and hinting. The smallest pixel size that makes sense is 1 whole pixel.

**Reply by cybershape** [reply]
Thank you for the quick reply!

May I add one more question: How is wezterm handle the fractional part of line_height config? Is it always rounded to integer (full pixel) ?

**Reply by wez** [reply]
Fractional cell metrics are rounded up to the full integer pixel value
